acasclient/validation.py

```diff
@@ -173,4 +180,64 @@ def get_validation_response(validation_result, ls_thing=None, commit=False, tran
     }
     if ls_thing:
         resp_dict['results']['thing'] = ls_thing
-    return resp_dict
\ No newline at end of file
+    return resp_dict
+
+
+def _get_html_summary(errors, warnings, summaries, commit=False) -> str:
+    """
+    Format HTML summary for the validation result.
+    """
+    # Lay out HTML template structure
+    HTML_SUMMARY_TEMPLATE = """<p>{instructions}</p>
+    {errors_block}
+    {warnings_block}
+    {summary_block}
+    """
+    ERRORS_TEMPLATE = """<h4 style=\"color:red\">Errors: {count} </h4>
+    <ul>{message_list}</ul>
+    """
+    WARNINGS_TEMPLATE = """<h4>Warnings: {count} </h4>
+    <p>Warnings provide information on issues found in the upload file. You can proceed with warnings; however, it is recommended that, if possible, you make the changes suggested by the warnings and upload a new version of the file by using the 'Back' button at the bottom of this screen.<p>
+    <ul>{message_list}</ul>"""
+    SUMMARY_TEMPLATE = """<h4>Summary</h4>
+    <p>Information:</p>
+    <ul>{message_list}</ul>"""
+    MESSAGE_TEMPLATE = """<li>{message}</li>"""
+
+    # Set up variables
+    instructions = ''
+    errors_block = ''
+    warnings_block = ''
+    summary_block = ''
+    if errors:
+        error_messages = '\n'.join(
+            [MESSAGE_TEMPLATE.format(message=error) for error in errors])
+        errors_block = ERRORS_TEMPLATE.format(
+            count=len(errors), message_list=error_messages)
+    if warnings:
+        warnings_messages = '\n'.join(
+            [MESSAGE_TEMPLATE.format(message=warning) for warning in warnings])
+        warnings_block = WARNINGS_TEMPLATE.format(
+            count=len(warnings), message_list=warnings_messages)
+    if summaries:
+        summary_messages = '\n'.join(
+            [MESSAGE_TEMPLATE.format(message=summary) for summary in summaries])
+        summary_block = SUMMARY_TEMPLATE.format(message_list=summary_messages)
+    if commit:
+        instructions = 'Upload completed.'
+        # hide warnings and errors if we've already committed
+        errors_block = ''
+        warnings_block = ''
+    elif errors:
+        instructions = "Please fix the following errors and use the 'Back' button at the bottom of this screen to upload a new version of the file."
```

Comment:
With my suggestion it may not be just a file change:

change:
```
upload a new version of the file
```

to:

```
upload a new version of the data
```